Formula/terraform.rb

```diff
@@ -52,21 +54,27 @@ def install
       buildpath.install "stringer"
     end
 
+    cd "src/github.com/kisielk/errcheck" do
+      system "go", "build"
+      buildpath.install "errcheck"
+    end
+
     cd terrapath do
       # v0.6.12 - source contains tests which fail if these environment variables are set locally.
       ENV.delete "AWS_ACCESS_KEY"
       ENV.delete "AWS_SECRET_KEY"
 
-      terraform_files = `go list ./...`.lines.map { |f| f.strip unless f.include? "/vendor/" }.compact
-      system "go", "test", *terraform_files
+      # Runs format check and test suite via makefile
+      ENV.deparallelize { system "make", "test", "vet" }
```

Comment:
Doing `-j20` or `-j` may help reveal this problem more easily; generally any time a `Makefile` has a parellelisation issue like this it's considered a bug. That said, I appreciate you've spent a bunch of time here so I'll pull this as-is but a report upstream would still be great (and I'm happy to help, there).